dpdata/abacus/scf.py

```diff
@@ -109,6 +109,96 @@ def get_cell(geometry_inlines):
     cell = celldm * np.array(cell)
     return celldm, cell
 
+def parse_stru_pos(pos_line):
+    '''
+  The content in atom position block:
+  - `m` or NO key word: three numbers, which take value in 0 or 1, control how the atom move in geometry relaxation calculations. In example below, the numbers `0 0 0` following the coordinates of the first atom means this atom are *not allowed* to move in all three directions, and the numbers `1 1 1` following the coordinates of the second atom means this atom *can* move in all three directions.
+  - `v` or `vel` or `velocity`: set the three components of initial velocity of atoms in geometry relaxation calculations(e. g. `v 1.0 1.0 1.0`).
+  - `mag` or `magmom` : set the start magnetization for each atom. In colinear case only one number should be given. In non-colinear case one have two choice:either set one number for the norm of magnetization here and specify two polar angle later(e. g. see below), or set three number for the xyz commponent of magnetization here (e. g. `mag 0.0 0.0 1.0`). Note that if this parameter is set, the initial magnetic moment setting in the second line will be overrided.
+    - `angle1`: in non-colinear case, specify the angle between c-axis and real spin, in angle measure instead of radian measure
+    - `angle2`: in non-colinear case, specify angle between a-axis and real spin in projection in ab-plane , in angle measure instead of radian measure
+
+      e.g.:
+
+      ```
+      Fe
+      1.0
+      2
+      0.0 0.0 0.0 m 0 0 0 mag 1.0 angle1 90 angle2 0 cs 0 0 0
+      0.5 0.5 0.5 m 1 1 1 mag 1.0 angle1 90 angle2 180
+  - `cs` or `constrain`: three numbers, which take value in 0 or 1, control the spin constraint of the atom.
+  - `lambda`: three numbers, control the lambda of the atom.    
+    '''
+    sline = pos_line.split()
+    pos = [float(i) for i in sline[:3]]
+    move = None
+    velocity = None
+    magmom = None
+    angle1 = None
+    angle2 = None
+    constrain = None
+    lambda1 = None
+    if len(sline) > 3:
+        mag_list = []
+        velocity_list = []
+        move_list = []
+        angle1_list = []
+        angle2_list = []
+        constrain_list = []
+        lambda_list = []
+        label = "move"
+        for i in range(3,len(sline)):
+            # firstly read the label
+            if sline[i] == "m":
+                label = "move"
+            elif sline[i] in ["v","vel","velocity"]:
+                label = "velocity"
+            elif sline[i] in ["mag","magmom"]:
+                label = "magmom"
+            elif sline[i] == "angle1":
+                label = "angle1"
+            elif sline[i] == "angle2":
+                label = "angle2"
```

Comment:
line not covered by UT. please check codecov for all the lines that are not covered by UT.